services/blockchain-indexer/shared/dataService/business/tokens.js

```diff
@@ -133,46 +124,33 @@ const getTopLiskAddresses = async (params) => {
 	return topLiskAddresses;
 };
 
-const getSupportedTokens = async (params) => {
-	const supportedTokensList = {
+const getTokensSummary = async (params) => {
+	const summary = {
 		data: {
-			supportedTokens: [],
 		},
 		meta: {},
 	};
 
-	const response = await requestConnector('token_getSupportedTokens');
-
-	supportedTokensList.data.supportedTokens = await BluebirdPromise.map(
-		response.tokenIDs,
-		async (tokenID) => {
-			const tokenMetadataResponse = await getTokenMetadataByID(tokenID);
-			const [tokenMetadata = {}] = tokenMetadataResponse.data || [];
-
-			return {
-				tokenID,
-				symbol: tokenMetadata.symbol,
-				name: tokenMetadata.tokenName,
-			};
-		},
-		{ concurrency: response.tokenIDs.length },
-	);
+	const escrowedAmount = await fetchInfoFromConnector('token_getEscrowedAmounts', params.offset, params.limit);
+	const supportedTokens = await fetchInfoFromConnector('token_getSupportedTokens', params.offset, params.limit);
+	const totalSupply = await fetchInfoFromConnector('token_getTotalSupply', params.offset, params.limit);
 
-	const total = supportedTokensList.data.supportedTokens.length;
-	supportedTokensList.data.supportedTokens = supportedTokensList.data.supportedTokens
-		.slice(params.offset, params.offset + params.limit);
+	const escrowedAmountResponse = await populateTokenMetaInfo(escrowedAmount);
+	const supportedTokensResponse = await populateTokenMetaInfo(supportedTokens);
+	const totalSupplyResponse = await populateTokenMetaInfo(totalSupply);
```

Comment:
Please consider creating a set of tokenIDs from all the above three responses and fetching the information only once from the app registry with every method invocation and reusing it to populate the responses.

Also, please add a fallback to query from the Mainnet Lisk Service endpoint in case the enduser decides not to run the AppRegistry microservice.